### ISSUE TITLE
Docstrings' format

### ISSUE THREAD

**Issue by mfocko** [packit/ogr#532]
Tools that we use for autogeneration of documentation support mainly markdown (also optionally numpy or google style of docstrings). We should transition from our current sphinx-style of comments to one of those.

- [ ] decide between markdown, numpy or google style
   - [ ] update GitHub action for generating of documentation if necessary
- [ ] update docstrings to adhere to the chosen style/format

Follow-up of #531

**Comment by csomh** [comment]
I have a prefference for Google Style doc-strings mainly because it's much easier on the eyes, which is important for those who are stuck in the past century and still using vim for editing.

But if I recall it correctly Sphinx-style was preffered, b/c that is the default (?) in PyCharm, which is used by most of the team (for reasons I cannot comprehend till this day, but that's another story). Anyways, quick search in the PyCharm docs [reveals](https://www.jetbrains.com/help/pycharm/type-syntax-for-docstrings.html) that Google-style might be also supported, although they call it "legacy" for some reason :man_shrugging: 

Maybe @lachmanfrantisek could check if this is true :slightly_smiling_face:

**Comment by lachmanfrantisek** [comment]
> Maybe @lachmanfrantisek could check if this is true

Yes, one reason is probably PyCharm, the other one is that using Sphinx was more common at that time (?).

Using markdown would be more natural but I don't know about any standard/integration/checker/...

So, if both pdoc and PyCharm supports both google and numpy, I would use one of these.

Personally, I like the google style more since we use type annotations in the code.

```
    Args:
        param1: The first parameter.
        param2: The second parameter.
```

instead of:

```
    Parameters
    ----------
    param1
        The first parameter.
    param2
        The second parameter.
```

**Comment by mfocko** [comment]
there is some progress https://github.com/mfocko/ogr/tree/documentation
